### DISCUSSION TITLE
关于VS报错

### DISCUSSION THREAD

**Discussion by System-BXV** [hooke007/mpv_PlayKit#109] [mpv 通用]
我尝试将mpv_lazy的补帧移动到archlinux上，但事实上无法解决这个报错，除了mvtools以外其他补帧全部无法执行。我尝试了重装VS和mpv但是都没有效果

依赖库:
archlinuxcn/mpv-git 0.34.0_220_gb15b3f6711-1
core/python 3.10.2-1
community/vapoursynth R57-2
extra/ffmpeg 2:5.0-4
coummunity/yt-dlp 2022.02.04-2
extra/tcl 8.6.12-3
extra/tk 8.6.12-1
aur/vapoursynth-plugin-mvtools 23-1
aur/vapoursynth-plugin-svpflow1 4.2.0.142-2
aur/vapoursynth-plugin-svpflow2-bin 4.3.0.168-1
aur/vapoursynth-plugin-temporalsoften2-git v1.1.ga133670-1
报错日志一:
```
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Script evaluation failed:
[vapoursynth] Python exception: SVSmoothFps: unable to determine source frame rate
[vapoursynth] 
[vapoursynth] Traceback (most recent call last):
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2832, in vapoursynth._vpy_evaluate
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2833, in vapoursynth._vpy_evaluate
[vapoursynth]   File "/home/bxv/.config/mpv/vs/svpflow-8b.vpy", line 32, in <module>
[vapoursynth]     interpolate(video_in).set_output()
[vapoursynth]   File "/home/bxv/.config/mpv/vs/svpflow-8b.vpy", line 25, in interpolate
[vapoursynth]     smooth = core.svp2.SmoothFps(
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2580, in vapoursynth.Function.__call__
[vapoursynth] vapoursynth.Error: SVSmoothFps: unable to determine source frame rate
[vapoursynth] 
mpv: ../video/filter/vf_vapoursynth.c:598: destroy_vs: Assertion `!p->in_node_active' failed.
已放弃 (核心已转储)
```
报错日志二：
```
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Frame requested during init! This is unsupported.
[vapoursynth] Returning black dummy frame with 0 duration.
[vapoursynth] Script evaluation failed:
[vapoursynth] Python exception: SVSmoothFps: unable to init GPU-based renderer - code 255
[vapoursynth] 
[vapoursynth] Traceback (most recent call last):
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2832, in vapoursynth._vpy_evaluate
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2833, in vapoursynth._vpy_evaluate
[vapoursynth]   File "/home/bxv/.config/mpv/vs/svpflow-2x.vpy", line 29, in <module>
[vapoursynth]     clip = core.svp2.SmoothFps(clip, svps["clip"], svps["data"], svpv["clip"], svpv["data"],
[vapoursynth]   File "src/cython/vapoursynth.pyx", line 2580, in vapoursynth.Function.__call__
[vapoursynth] vapoursynth.Error: SVSmoothFps: unable to init GPU-based renderer - code 255
[vapoursynth] 
[vapoursynth] could not init VS
(Paused) AV: 00:00:00.000 / 00:23:40.074 (0%) A-V:  0.000
Disabling filter vapoursynth.01 because it has failed.
```

**Comment by hooke007** [comment]
> unable to init GPU-based renderer-code 255

svpflow-2x里的这应该是个opencl错误，可以尝试先禁用svp的显卡加速，如果有效说明原因正确，然后再尝试补足opencl的相关依赖？

> unable to determine source frame rate

至于8b里这个看着像是视频源的问题，这个视频在win环境下8b能否正常工作

**Reply by System-BXV** [reply]
非常感谢你的答复，我在补足相关依赖后就能正常工作了，但是在调整进度或者中途更换插帧模板时会卡顿导致崩溃后报错
报错日志:
[load_plus] 自动队列 追加（前）[LowPower-Raws] Yosuga no Sora - 01 (BD 1080P x264 FLACx3).mkv 
AV: 00:00:02.127 / 00:01:32.010 (2%) A-V:  0.011 DS: 5.792/30 Dropped: 1 Cache: 89s/81MB
[autoconvert] Converting nv12 -> yuv420p
AV: 00:00:02.252 / 00:01:32.010 (2%) A-V:  0.085 DS: 5.589/32 Dropped: 2 Cache: 87s/80MB
VO: [gpu] 1920x1080 yuv420p
段错误 (核心已转储)00:01:32.010 (4%) A-V:  0.046 DS: 1.590/13 Dropped: 38 Cache: 87s/80MB

**Reply by hooke007** [reply]
这个只是控制台的操作日志，需要 `--log-file` 才能知道mpv的内部事件

**Reply by System-BXV** [reply]
[log.txt ](https://github.com/hooke007/MPV_lazy/files/8456532/log.txt)
这个日志的最后操作是卸载vpy配置

**Reply by hooke007** [reply]
只是svpflow才有这个问题还是所有vapoursynth滤镜都有这个问题？但是无论如何这个问题我没辙

**Reply by System-BXV** [reply]
az,是这个样子嘛。除了mvtools以外的那几个补帧脚本都会这样

**Reply by hooke007** [reply]
只是svpflow的话那可能是它的独有bug，win这里是没有办法才用的上古版本。。。不过svp linux本身不是免费的吗，linux也没有便携这个概念，要不直接用svp manager?

**Reply by System-BXV** [reply]
使用svp manager也会崩溃,emmmm 奇怪
[文本文件.txt ](https://github.com/hooke007/MPV_lazy/files/8458156/default.txt)

**Reply by System-BXV** [reply]
但是修改播放速度不会崩溃，emmm

**Reply by System-BXV** [reply]
[active.log](https://github.com/hooke007/MPV_lazy/files/8458169/active.log)
这是我的svp4输出的日志

**Reply by hooke007** [reply]
整个干净的配置文件也能稳定复现的话只能去svp论坛那边求助了

**Reply by System-BXV** [reply]
好的，感谢你的回复
